### PR TITLE
Wire in the BucketExecutor.

### PR DESCRIPTION
**Pull request by baldersheim**
@toregge and @geirst PR

### CODE REVIEW

**Review by toregge — APPROVED** [review]

File: searchcore/src/vespa/searchcore/proton/server/lid_space_compaction_job_take2.cpp [hunk 1]
```diff
@@ -44,16 +47,20 @@ CompactionJob::moveDocument(const search::DocumentMetaData & metaThen, std::shar
     if (metaNow.lid != metaThen.lid) return;
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/server/lid_space_compaction_job_take2.cpp:47]: __scanItr->getMetaData()_ is now called by the bucket executor thread. Is that really safe? We are looking into a structure that is traditionally updated by the docdb writer thread. @toregge I would like your opinion here as well.

File: searchcore/src/vespa/searchcore/proton/server/maintenance_jobs_injector.cpp [hunk 2]
```diff
@@ -34,15 +36,27 @@ injectLidSpaceCompactionJobs(MaintenanceController &controller,
...
-                config.getLidSpaceCompactionConfig(),
-                *lidHandler, opStorer, fbHandler,
-                diskMemUsageNotifier,
-                config.getBlockableJobConfig(),
-                clusterStateChangedNotifier,
-                (calc ? calc->nodeRetired() : false));
+
+        IMaintenanceJob::UP job;
+        if (config.getLidSpaceCompactionConfig().useBucketExecutor()) {
+            job = std::make_unique<LidSpaceCompactionJob>(
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/server/maintenance_jobs_injector.cpp:43]: _lidspace::CompactionJob_ should be instantiated here, or we must negate the expression in the if statement.

File: searchcore/src/vespa/searchcore/proton/server/lid_space_compaction_job_take2.cpp (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/lid_space_compaction_job_take2.cpp:47]: Yes, I came to the same conclusion myself in #16190

File: searchcore/src/vespa/searchcore/proton/server/maintenance_jobs_injector.cpp (hunk 2 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/maintenance_jobs_injector.cpp:43]: Correct, fixed #16190
Very good observations, both you and factory came to the same conclusion :)

**Pull request merged**